### ISSUE TITLE
Test build failure on stackage nightly

### ISSUE THREAD

**Issue by mihaimaruseac** [rudymatela/speculate#1]
```
Preprocessing test suite 'expr' for speculate-0.2.6..
Building test suite 'expr' for speculate-0.2.6..

<no location info>: warning: [-Wmissing-home-modules]
    These modules are needed for compilation but not listed in your .cabal file's other-modules: Test.Speculate
                                                                                                 Test.Speculate.Args
                                                                                                 Test.Speculate.CondReason
                                                                                                 Test.Speculate.Engine
                                                                                                 Test.Speculate.Expr
                                                                                                 Test.Speculate.Expr.Canon
                                                                                                 Test.Speculate.Expr.Core
                                                                                                 Test.Speculate.Expr.Equate
                                                                                                 Test.Speculate.Expr.Ground
                                                                                                 Test.Speculate.Expr.Instance
                                                                                                 Test.Speculate.Expr.Match
                                                                                                 Test.Speculate.Pretty
                                                                                                 Test.Speculate.Reason
                                                                                                 Test.Speculate.Reason.Order
                                                                                                 Test.Speculate.Report
                                                                                                 Test.Speculate.Sanity
                                                                                                 Test.Speculate.SemiReason
                                                                                                 Test.Speculate.Utils
                                                                                                 Test.Speculate.Utils.Class
                                                                                                 Test.Speculate.Utils.Colour
                                                                                                 Test.Speculate.Utils.Digraph
                                                                                                 Test.Speculate.Utils.List
                                                                                                 Test.Speculate.Utils.Memoize
                                                                                                 Test.Speculate.Utils.Misc
                                                                                                 Test.Speculate.Utils.Ord
                                                                                                 Test.Speculate.Utils.PrettyPrint
                                                                                                 Test.Speculate.Utils.String
                                                                                                 Test.Speculate.Utils.Tiers
                                                                                                 Test.Speculate.Utils.Timeout
                                                                                                 Test.Speculate.Utils.Tuple
                                                                                                 Test.Speculate.Utils.Typeable
[ 1 of 33] Compiling Test.Speculate.Utils.Colour ( src/Test/Speculate/Utils/Colour.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Colour.o )
[ 2 of 33] Compiling Test.Speculate.Utils.List ( src/Test/Speculate/Utils/List.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/List.o )
[ 3 of 33] Compiling Test.Speculate.Utils.Class ( src/Test/Speculate/Utils/Class.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Class.o )
[ 4 of 33] Compiling Test.Speculate.Utils.Memoize ( src/Test/Speculate/Utils/Memoize.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Memoize.o )
[ 5 of 33] Compiling Test.Speculate.Utils.Ord ( src/Test/Speculate/Utils/Ord.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Ord.o )
[ 6 of 33] Compiling Test.Speculate.Utils.PrettyPrint ( src/Test/Speculate/Utils/PrettyPrint.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/PrettyPrint.o )
[ 7 of 33] Compiling Test.Speculate.Utils.String ( src/Test/Speculate/Utils/String.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/String.o )
[ 8 of 33] Compiling Test.Speculate.Utils.Misc ( src/Test/Speculate/Utils/Misc.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Misc.o )
[ 9 of 33] Compiling Test.Speculate.Utils.Tiers ( src/Test/Speculate/Utils/Tiers.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Tiers.o )
[10 of 33] Compiling Test.Speculate.Utils.Timeout ( src/Test/Speculate/Utils/Timeout.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Timeout.o )
[11 of 33] Compiling Test.Speculate.Utils.Tuple ( src/Test/Speculate/Utils/Tuple.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Tuple.o )
[12 of 33] Compiling Test.Speculate.Utils.Typeable ( src/Test/Speculate/Utils/Typeable.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Typeable.o )
[13 of 33] Compiling Test.Speculate.Utils ( src/Test/Speculate/Utils.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils.o )
[14 of 33] Compiling Test.Speculate.Utils.Digraph ( src/Test/Speculate/Utils/Digraph.hs, dist/build/expr/expr-tmp/Test/Speculate/Utils/Digraph.o )
[15 of 33] Compiling Test.Speculate.Expr.Core ( src/Test/Speculate/Expr/Core.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr/Core.o )
[16 of 33] Compiling Test.Speculate.Expr.Match ( src/Test/Speculate/Expr/Match.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr/Match.o )
[17 of 33] Compiling Test.Speculate.Expr.Instance ( src/Test/Speculate/Expr/Instance.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr/Instance.o )
[18 of 33] Compiling Test.Speculate.Expr.Equate ( src/Test/Speculate/Expr/Equate.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr/Equate.o )
[19 of 33] Compiling Test.Speculate.Expr.Ground ( src/Test/Speculate/Expr/Ground.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr/Ground.o )
[20 of 33] Compiling Test.Speculate.Expr.Canon ( src/Test/Speculate/Expr/Canon.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr/Canon.o )
[21 of 33] Compiling Test.Speculate.Expr ( src/Test/Speculate/Expr.hs, dist/build/expr/expr-tmp/Test/Speculate/Expr.o )
[22 of 33] Compiling Test.Speculate.Sanity ( src/Test/Speculate/Sanity.hs, dist/build/expr/expr-tmp/Test/Speculate/Sanity.o )
[23 of 33] Compiling Test.Speculate.Reason.Order ( src/Test/Speculate/Reason/Order.hs, dist/build/expr/expr-tmp/Test/Speculate/Reason/Order.o )
[24 of 33] Compiling Test.Speculate.Reason ( src/Test/Speculate/Reason.hs, dist/build/expr/expr-tmp/Test/Speculate/Reason.o )
[25 of 33] Compiling Test.Speculate.SemiReason ( src/Test/Speculate/SemiReason.hs, dist/build/expr/expr-tmp/Test/Speculate/SemiReason.o )
[26 of 33] Compiling Test.Speculate.CondReason ( src/Test/Speculate/CondReason.hs, dist/build/expr/expr-tmp/Test/Speculate/CondReason.o )
[27 of 33] Compiling Test.Speculate.Pretty ( src/Test/Speculate/Pretty.hs, dist/build/expr/expr-tmp/Test/Speculate/Pretty.o )
[28 of 33] Compiling Test.Speculate.Engine ( src/Test/Speculate/Engine.hs, dist/build/expr/expr-tmp/Test/Speculate/Engine.o )
[29 of 33] Compiling Test.Speculate.Args ( src/Test/Speculate/Args.hs, dist/build/expr/expr-tmp/Test/Speculate/Args.o )
[30 of 33] Compiling Test.Speculate.Report ( src/Test/Speculate/Report.hs, dist/build/expr/expr-tmp/Test/Speculate/Report.o )
[31 of 33] Compiling Test.Speculate   ( src/Test/Speculate.hs, dist/build/expr/expr-tmp/Test/Speculate.o )
[32 of 33] Compiling Test             ( tests/Test.hs, dist/build/expr/expr-tmp/Test.o )
[33 of 33] Compiling Main             ( tests/test-expr.hs, dist/build/expr/expr-tmp/Main.o )

<no location info>: warning: [-Wmissing-home-modules]
    These modules are needed for compilation but not listed in your .cabal file's other-modules: Test.Speculate
                                                                                                 Test.Speculate.Args
                                                                                                 Test.Speculate.CondReason
                                                                                                 Test.Speculate.Engine
                                                                                                 Test.Speculate.Expr
                                                                                                 Test.Speculate.Expr.Canon
                                                                                                 Test.Speculate.Expr.Core
                                                                                                 Test.Speculate.Expr.Equate
                                                                                                 Test.Speculate.Expr.Ground
                                                                                                 Test.Speculate.Expr.Instance
                                                                                                 Test.Speculate.Expr.Match
                                                                                                 Test.Speculate.Pretty
                                                                                                 Test.Speculate.Reason
                                                                                                 Test.Speculate.Reason.Order
                                                                                                 Test.Speculate.Report
                                                                                                 Test.Speculate.Sanity
                                                                                                 Test.Speculate.SemiReason
                                                                                                 Test.Speculate.Utils
                                                                                                 Test.Speculate.Utils.Class
                                                                                                 Test.Speculate.Utils.Colour
                                                                                                 Test.Speculate.Utils.Digraph
                                                                                                 Test.Speculate.Utils.List
                                                                                                 Test.Speculate.Utils.Memoize
                                                                                                 Test.Speculate.Utils.Misc
                                                                                                 Test.Speculate.Utils.Ord
                                                                                                 Test.Speculate.Utils.PrettyPrint
                                                                                                 Test.Speculate.Utils.String
                                                                                                 Test.Speculate.Utils.Tiers
                                                                                                 Test.Speculate.Utils.Timeout
                                                                                                 Test.Speculate.Utils.Tuple
                                                                                                 Test.Speculate.Utils.Typeable
Linking dist/build/expr/expr ...
> /tmp/stackage-build14/speculate-0.2.6$ dist/build/expr/expr
*** Failed tests:[114]
```

This looks strange. I'll look into it too.

**Comment by rudymatela** [comment]
I'm sorry about the failure.  Thanks for reporting the issue.

I'll look into this later today.

**Comment by rudymatela** [comment]
I have disabled the test that is causing the issue, see 972aa21.

I have released [Speculate v0.2.7] on Hackage, [it builds fine on Travis].

When you have the time, you can revert fpco/stackage@c4eb8a6 to see if everything works now.

Again, thanks for reporting the issue.

[Speculate v0.2.7]: https://hackage.haskell.org/package/speculate-0.2.7
[it builds fine on Travis]: https://travis-ci.org/rudymatela/speculate/builds/263142588

**Comment by mihaimaruseac** [comment]
Thank you. It's fixed, indeed.
